ejb/src/main/java/org/wildfly/httpclient/ejb/HttpEJBInvocationBuilder.java

```diff
@@ -120,28 +120,33 @@ enum InvocationType {
 
     // helper methods
 
-    ClientRequest createRequest(final String mountPoint) {
-        ClientRequest clientRequest = new ClientRequest();
+    ClientRequest createRequest(final String prefix) {
+        final ClientRequest clientRequest = new ClientRequest();
+        clientRequest.setPath(getBeanRequestPath(prefix));
         if (invocationType == InvocationType.METHOD_INVOCATION) {
             clientRequest.setMethod(Methods.POST);
-            clientRequest.setPath(invokeBeanRequestPath(mountPoint));
             clientRequest.getRequestHeaders().add(Headers.ACCEPT, INVOCATION_ACCEPT + "," + EJB_EXCEPTION);
             if (invocationId != null) {
                 clientRequest.getRequestHeaders().put(INVOCATION_ID, invocationId);
             }
             clientRequest.getRequestHeaders().put(Headers.CONTENT_TYPE, INVOCATION.toString());
         } else if (invocationType == InvocationType.STATEFUL_CREATE) {
             clientRequest.setMethod(Methods.POST);
-            clientRequest.setPath(openBeanRequestPath(mountPoint));
             clientRequest.getRequestHeaders().put(Headers.CONTENT_TYPE, SESSION_OPEN.toString());
             clientRequest.getRequestHeaders().add(Headers.ACCEPT, EJB_EXCEPTION.toString());
         } else if(invocationType == InvocationType.CANCEL) {
             clientRequest.setMethod(Methods.DELETE);
-            clientRequest.setPath(cancelBeanRequestPath(mountPoint));
         }
         return clientRequest;
     }
 
+    private String getBeanRequestPath(final String prefix) {
```

Comment:
Here you are introducing a new path in the workflow of the code that is unnecessary. Test the invocation type only once. Can it be a switch case? I haven't checked the details for this.

Reply:
This is fixed later in this PR.